### DISCUSSION TITLE
check if headphones are inserted

### DISCUSSION THREAD

**Discussion by basbebe** [FelixKratz/SketchyBar#161] [Q&A] [answered]
Where would I check if headphones were inserted and audio output set to them?

**Comment by zcag** [comment, marked as answer]
You can use a custom script, i personally use this cli tool: https://github.com/deweller/switchaudio-osx

**Reply by basbebe** [reply]
This looks promising!

Do you know of a way to get a notification / trigger an event when Headphones are inserted or Audio input source changes?

**Reply by zcag** [reply]
I was just checking that :) will try the `AppleSelectedInputSourcesChangedNotification` event when i have the chance(i have no headphones with me to test atm)
Currently i have the freq of 3 for the script without any triggers.

**Reply by FelixKratz** [reply]
I believe the `AppleSelectedInputSourcesChangedNotification` is only for keyboard changes sadly, I don't know if there exists a 
notification when headphones are inserted. If you find one, be sure to share it here: https://github.com/FelixKratz/SketchyBar/discussions/151
